cloud/aws/services/cloudformation/bootstrap.go

```
@@ -0,0 +1,341 @@
+					"elasticloadbalancing:CreateListener",
+					"elasticloadbalancing:CreateTargetGroup",
+					"elasticloadbalancing:DeleteListener",
+					"elasticloadbalancing:DeleteTargetGroup",
+					"elasticloadbalancing:DescribeListeners",
+					"elasticloadbalancing:DescribeLoadBalancerPolicies",
+					"elasticloadbalancing:DescribeTargetGroups",
+					"elasticloadbalancing:DescribeTargetHealth",
+					"elasticloadbalancing:ModifyListener",
+					"elasticloadbalancing:ModifyTargetGroup",
+					"elasticloadbalancing:RegisterTargets",
+					"elasticloadbalancing:SetLoadBalancerPoliciesOfListener",
+					"iam:CreateServiceLinkedRole",
+					"kms:DescribeKey",
+				},
+			},
+		},
+	}
+}
+
+// From https://github.com/kubernetes/cloud-provider-aws
+func cloudProviderNodeAwsPolicy() *iam.PolicyDocument {
+	return &iam.PolicyDocument{
+		Version: iam.CurrentVersion,
+		Statement: []iam.StatementEntry{
+			{
+				Effect:   iam.EffectAllow,
+				Resource: iam.Resources{"*"},
+				Action: iam.Actions{
+					"ec2:DescribeInstances",
+					"ec2:DescribeRegions",
+					"ecr:GetAuthorizationToken",
+					"ecr:BatchCheckLayerAvailability",
+					"ecr:GetDownloadUrlForLayer",
+					"ecr:GetRepositoryPolicy",
+					"ecr:DescribeRepositories",
+					"ecr:ListImages",
+					"ecr:BatchGetImage",
+				},
+			},
+		},
+	}
+}
+
+// CreateBootstrapStack creates a AWS CloudFormation stack to initialise
+// IAM policy
+func CreateBootstrapStack() error {
+	stackName := "cluster-api-provider-aws-sigs-k8s-io"
+	sess := session.New()
+	svc := cfn.New(sess)
+
+	template := BootstrapTemplate()
+	yaml, err := template.YAML()
+	if err != nil {
+		return errors.Wrap(err, "failed to generate AWS CloudFormation YAML")
+	}
+
+	input := &cfn.CreateStackInput{
+		Capabilities: aws.StringSlice([]string{cfn.CapabilityCapabilityIam, cfn.CapabilityCapabilityNamedIam}),
+		TemplateBody: aws.String(string(yaml)),
+		StackName:    aws.String(stackName),
+	}
+
+	_, err = svc.CreateStack(input)
```

Looks like this error is being ignored, isn't it?

---

Yup. Sorted, thanks.